### PR TITLE
 Improved resource id creation

### PR DESCRIPTION
**Pull request by annando**
During the work at PR https://github.com/friendica/friendica/pull/4479 i realized that the resource id creation for photos wasn't optimal. Using a hash of an unique id isn't a very good idea.

The old function is now replaced with a new one that used the standard PHP function for unique ids. The function had been altered to support even more entropy in the same string length, so we don't need a check if that id had been used before.

Attention: This PR can only be merged, **after** PR https://github.com/friendica/friendica/pull/4479 had been merged.

### CODE REVIEW

**Comment by MrPetovan** [comment]
Blocked by #4479

**Comment by MrPetovan** [comment]
Why is it depending on #4479?

**Comment by annando** [comment]
When I created that branch I created it based on the other branch. I prefer not to change branches when the database versions differ between the current branch and "develop".

File: boot.php [hunk 1]
```diff
@@ -854,11 +854,13 @@ function check_addons(App $a)
 	return;
 }
 
-function get_guid($size = 16, $prefix = "")
+function get_guid($size = 16, $prefix = '')
 {
-	if ($prefix == "") {
+	if (is_bool($prefix)) {
```
Comment (MrPetovan) [inline boot.php]: If `$prefix` is `true`, this function should generate a default prefix as well.

**Comment by MrPetovan** [comment]
> When I created that branch I created it based on the other branch. I prefer not to change branches when the database versions differ between the current branch and "develop".

Watch out for changes you added to the base branch that don't appear here, you'll probably need to rebase this branch onto the original branch at some point:

```
git rebase db-structure better-resource-id
git push --force
```

**Comment by annando** [comment]
I will wait for https://github.com/friendica/friendica/pull/4479 to be merged. This is the safest way.

**Comment by MrPetovan** [comment]
This is equivalent than solving the merge conflict later, but it is up to you.

File: boot.php (hunk 1 above)
Comment (annando) [reply boot.php]: Done.

**Pull request merged**